tests/unit/utils/test_cast.py

```diff
@@ -166,6 +166,102 @@ def test_all_union_casts_fail() -> None:
         ("", str, ""),
         ("", int | str, ""),
         ("", Union[int, str], ""),
+        # JSON parsing tests - lists
```

Comment:
@njbrake I wonder if we are at the point where it would be worth trying https://hypothesis.readthedocs.io/en/latest/